Amplify/Categories/DataStore/Query/ModelKey.swift

```diff
@@ -32,24 +32,42 @@ import Foundation
 ///     post.content != nil
 /// })
 /// ```
-public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {}
+public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {
+    var modelName: String { get }
+}
+
+extension ModelKey {
+    public var modelName: String { "" }
+}
 
 extension CodingKey where Self: ModelKey {
 
+    var columnName: String {
+        guard let modelSchema: ModelSchema = ModelRegistry.modelSchema(from: modelName) else {
+            return stringValue
```

Comment:
warn level log to hint customers that they should upgrade their CLI and regenerate their code

Reply:
I don't know if there is better way of putting a log than printing here since class `CodingKey` does not support extension on `DefaultLogger`.

Reply:
would conforming `ModelKey` to `DefaultLogger` help?

Reply:
Also tried, since `ModelKey` is a protocol, it is not working.

Error message: 
```
Extension of protocol 'ModelKey' cannot have an inheritance clause
```